imports/lib/schemas/customTypes.ts

```diff
@@ -38,7 +38,7 @@ export const allowedEmptyString = z.string();
 // always return values (e.g. createdTimestamp will always be a Date, it'll just
 // get set on insertion and not on subsequent updates).
 
-export const stringId = z.string().regex(Id).optional()
+export const stringId = nonEmptyString.optional()
```

Comment:
I wonder if it makes sense to have `Model` continue assuming & enforcing Meteor-style `_id` keys, but to also have an `ExplicitlyKeyedModel` where the `_id` field is not optional.  Then we could enforce the "does `_id` look like a `Random.id()`" restriction in the relevant places for the majority of collections, while allowing things like `BlobMappings` or singleton collections to demand the caller specify the `_id` everywhere we want to interact with the collection.

Thoughts?

Reply:
Yeah, I had been kind of thinking about this, because it would be nice if (e.g.) `DriveActivityLatest` had its `_id` type declared as a `z.literal`, since it's a fixed value, and I came up with a way to do it. Pushing the new code in a sec, but the basic idea is that `stringId` remains a strict `z.string().regex(Id)`, but instead allow manually passing in a different schema for `_id` to the `Model` constructor.